nucypher/network/middleware.py

```diff
@@ -113,7 +116,10 @@ def send_work_order_payload_to_ursula(self, work_order):
 
     def node_information(self, host, port, certificate_filepath):
         endpoint = "https://{}:{}/public_information".format(host, port)
-        return requests.get(endpoint, verify=certificate_filepath, timeout=2)
+        response = requests.get(endpoint, verify=certificate_filepath, timeout=2)
+        if not response.status_code == 200:
+            raise RuntimeError("Got a bad response: {}".format(response))
```

Comment:
Besides my own desire for semantics, this isn't just a _bad response_. It can also be a 404 or any other status code that represents something we didn't intend to happen. Perhaps something better to say is, "Got an unexpected response: "?

Completely up to your discretion, though. :)